datasets/mx/governors/crawler.py

```diff
@@ -0,0 +1,124 @@
+    # We will then use a regex to extract the name, state, start_date and end_date
+
+    regex_pattern = r'col-xs-12 col-sm-6 col-md-4 pb15 mix (.*?) \[(.*?)\] (\d{2}/\d{2}/\d{4}) (\d{2}/\d{2}/\d{4})'
+
+    match = re.match(regex_pattern, input_html.get('class'))
+
+    if match is None:
+        context.log.warning("Unable to extract information from HTML element")
+
+    name, state, start_date, end_date = match.group(1), match.group(2), match.group(3), match.group(4)
+
+    person = context.make("Person")
+    person.id = context.make_id(name)
+
+    # The names in the card always have a title as the first word, for example,
+    # Ing. Carlos Lozano de la Torre, where Ing. stands for Engineer
+    # We will add two name propreties, one with the "full" name
+    # and other with the clean name. The clean name
+    # can be retrived from the url to that governors page
+    person.add("name", name)
+    clean_name = (link_governor_page
+                  .replace("https://www.conago.org.mx/miembros/detalle/", "")
+                  .replace("-", " ")
+                  .title())
+    person.add("name", clean_name)
+
+    birth_place, birth_date = extract_birth_place_and_date(link_governor_page, context)
+
+    if birth_place is not None:
+        person.add("birthPlace", birth_place)
+        person.add("birthDate", h.parse_date(birth_date, formats=["%d/%m/%Y"])[0])
+
+    name_of_position = "Governor of " + state.title()
+    position = h.make_position(context, name_of_position, country="mx")
+    categorisation = categorise(context, position, is_pep=True)
+
+    occupancy = h.make_occupancy(
+            context,
+            person,
+            position,
+            no_end_implies_current=True,
+            start_date=h.parse_date(start_date, formats=["%d/%m/%Y"])[0],
+            end_date=h.parse_date(end_date, formats=["%d/%m/%Y"])[0],
+            categorisation=categorisation,
+        )
+
+
+    # If occupancy is None, then it is not a PEP
+    if occupancy is not None:
+        context.emit(person, target=True)
+        context.emit(position)
+        context.emit(occupancy)
+
+
+    
+
+def crawl(context: Context):
+    """
+    Entrypoint to the crawler.
+
+    The crawler works by first downloading the HTML of the page
+    and then iterating for each div that represents a Mexico State Governor,
+    to create the entities and positions.
+
+    :param context: The context object.
+    """
+
+    response = context.fetch_html(context.data_url)
+
+    if response is None:
+        context.log.error("Error while fetching the HTML")
+        return
+
+    path_to_cards = '//*[@class="containerMixitup"]/div/div'
+    path_to_links = '//*[@class="containerMixitup"]/div/div/div/div/div/div[1]/a'
+    for (item, a_tag) in zip(response.xpath(path_to_cards), response.xpath(path_to_links)):
+        crawl_item(item, a_tag.get('href'), context)
```

Comment:
This is cool but if feels a little too clever and there's the slightest chance it misaligns data one day. I think a little safer and more completely obvious in how it works would simply to do the link xpath on the card element inside the loop

Reply:
I will do that